coreax/score_matching.py

```diff
@@ -60,7 +60,7 @@
 _RandomGenerator = Callable[[KeyArrayLike, Sequence[int], DTypeLike], Array]
 
 
-class ScoreMatching(ABC, eqx.Module):
```

Comment:
I don't get any test errors even without this change? (With the versions specified in the `uv.lock` on this branch.)

Reply:
Interesting, I think it fails if ABC is the first argument (like it was before) but not if it's the second one. I'm happy to add ABC as a second argument if there is a good reason for this?

Reply:
I was just following the example from equinox: https://docs.kidger.site/equinox/pattern/, who don't include ABC in the inheritance. I have tested that `@abstractmethod` is enough to enforce method implementation, I'm not sure if there is any other functionality we want. 

If it's just for the purpose of being explicit, I'm happy to include the explicit inheritance, although the decorator is fairly explicit. 

Anyway, this is a fairly insignificant change, so happy to go with your final call, but it's been a good discussion in case something like this crops up again.

Reply:
Having looked a little closer into the code, `eqx.Module` is itself an abstract class (its metaclass is a subclass of `ABCMeta`) so we shouldn't need to inherit from ABC or explicitly set the metaclass ourself.

I think we'd want to do at least one of: 
- renaming it to `AbstractScoreMatching` (this one is what the Equinox docs do, but renaming is a fairly broad change which might be a hassle wrt deprecation periods etc)
- Making it very obvious in the docstring (in the first line ideally) that it's an _abstract_ base class

to make it immediately obvious that it's an abstract class.